chainer/training/trigger.py

```diff
@@ -20,10 +20,14 @@ def __init__(self, period, unit):
         self.period = period
         assert unit == 'epoch' or unit == 'iteration'
         self.unit = unit
+        self.count = 0
 
     def __call__(self, trainer):
         """Decides whether the extension should be called on this iteration.
 
+        In order to support non-integer epoch intervals, it does not rely on
+        ``updater.iterator.is_new_epoch``.
```

Comment:
So this description is for developers of Chainer, not its users because it explains *how* it is implemented not *what* is implemented. Further, I do not think it is natural for users to use `updater.is_new_epoch` to realize this method.

So I think we need not to write it (or we write the motivation of the design a  note with slight detailed background).  For example, how about the followings?
"For developers: it would be natural at first sight to use updater.is_new_epoch to decide if the trigger fires or not when `unit` is 'epoch'. But we cannot judge solely from it if the period is floating values. That is why IntervalTrigger directly sees updater.epoch_detail and decides if it fires."